BuildRequires enhancement when test suite is present

mcyprian opened the pull request:
Detection of test suite based on data from setup.py is not reliable. Check if the archive contains tests/ directory was added. In case the test suite is detected, all the install requires are added to BuildRequires as well to make sure all the dependencies to run integration tests will be available in the Buildroot.

irushchyshyn commented on pyp2rpm/metadata_extractors.py:
```diff
@@ -584,7 +590,10 @@ def license(self):
     @property
     def has_test_suite(self):
-        return self.json_metadata.get('test_requires', False) is not False
+        test_dir = "{0}/tests".format(self.archive.top_directory, 'tests') in \
```
Can we move `test_dir` to be the property/method of `LocalMetadataExtractor` and reuse it for both cases?

Also if `test_dir` is a bool, it would be better to name it smth like `has_test_dir`.
Also the `format` method gets an extra useless argument `tests`.

irushchyshyn commented on pyp2rpm/metadata_extractors.py:
```diff
@@ -371,14 +371,18 @@ def runtime_deps(self):  # install_requires
             'tests_require']
+
+        if self.has_test_suite:
+            build_requires += self.metadata['install_requires']
```
Can we also move adding `self.metadata['tests_require']` here from the line above?
I will not change the behavior, but it would look more logical.

irushchyshyn replied:
Also why don't we add the same handling for `build_deps` in `WheelMetadataExtractor` to add `run_requires` if the test suite is present?

mcyprian commented:
Test suite detection was reimplemented to search for files with test_ prefix or _test suffix instead of tests directory. pytest does the similar thing when collecting test files. All the requested changes should be covered by two fixup commits.

irushchyshyn reviewed (approved):
Looks good 👍 
Please rebase and merge at will.

mcyprian committed:
Improve detection of test suite

Search for tests/ directory in top directory of archive,
include %check section if present

mcyprian committed:
Enhance BuildRequires when %check is present

Add all install_requires listed in setup.py to BuildRequires
when test suite is present.
